Added Enchanced Mushrooms compat for hanging mushroom feature.

Dariensg opened the pull request:
-If Enhanced Mushrooms is installed, replaces Brown Mushroom stem in GSC with their stem

Dariensg committed:
Added Mushroom Stem Compat

-If Enhanced Mushrooms is installed, replaces Brown Mushroom stem in GSC with their stem

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/world/gen/features/HangingGiantBrownMushroomFeature.java:
```diff
@@ -31,7 +33,11 @@ public boolean generate(ISeedReader world, ChunkGenerator chunk, Random random,
             for (int y = 0; y <= size; y++) {
-                world.setBlockState(pos.down(y), Blocks.MUSHROOM_STEM.getDefaultState(), 2);
+                if (DataUtil.isLoaded("enhanced_mushrooms")) {
+                    world.setBlockState(pos.down(y), ForgeRegistries.BLOCKS.getValue(new ResourceLocation("enhanced_mushrooms:brown_mushroom_stem")).getDefaultState(), 2);
```
```suggestion
                    world.setBlockState(pos.down(y), ForgeRegistries.BLOCKS.getValue(new ResourceLocation("enhanced_mushrooms", "brown_mushroom_stem")).getDefaultState(), 2);
```
Creating a resource location with this format is a bit inefficient, since it requires string parsing. This code gets called a lot in generation, so we have to optimize it in every way possible.

If you don't think it's silly, you could also add a private static field for the resource location to avoid making a new instance for it every time this code runs.

Hellion-Games replied:
I'll go ahead and make a private static field!

Dariensg committed:
Increase Efficiency

-Made ResourceLocation static, unparsed field to make it more efficient
-Moved grabbing of block from registry outside of for loop to increase efficiency
